pkg/controllers/node/termination/controller.go

```diff
@@ -104,11 +105,18 @@ func (c *Controller) Finalize(ctx context.Context, node *v1.Node) (reconcile.Res
 		}
 		return reconcile.Result{RequeueAfter: 1 * time.Second}, nil
 	}
-	// Be careful when removing this delete call in the Node termination flow
-	// This delete call is needed so that we ensure that we don't remove the node from the cluster
-	// until the full instance shutdown has taken place
-	if err := c.cloudProvider.Delete(ctx, nodeclaimutil.NewFromNode(node)); cloudprovider.IgnoreNodeClaimNotFoundError(err) != nil {
-		return reconcile.Result{}, fmt.Errorf("terminating cloudprovider instance, %w", err)
+	nodeClaims, err := nodeutils.GetNodeClaims(ctx, node, c.kubeClient)
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("deleting nodeclaims, %w", err)
+	}
+	for _, nodeClaim := range nodeClaims {
+		requeue, err := utils.EnsureTerminated(ctx, c.kubeClient, nodeClaim, c.cloudProvider)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		if requeue {
+			return reconcile.Result{Requeue: requeue}, nil
```

Comment:
I think we need to be careful about requeing with back-off here when the instance probably needs to be checked at a static cadence. This is going to cause us to check with exponential back-off and might result in higher latency for the termination